pynxtools/dataconverter/verify.py

```diff
@@ -0,0 +1,139 @@
+"""Verifies a nxs file"""
+import os
+import sys
+from typing import Dict, Optional, Union
+import xml.etree.ElementTree as ET
+import logging
+from h5py import File, Dataset, Group
+import click
+
+from pynxtools.dataconverter import helpers
+from pynxtools.dataconverter.template import Template
+from pynxtools.nexus import nexus
+
+logger = logging.getLogger(__name__)
+
+DEBUG_TEMPLATE = 9
+logger.setLevel(logging.INFO)
+logger.addHandler(logging.StreamHandler(sys.stdout))
+
+
+def _replace_group_names(class_map: Dict[str, str], path: str):
+    for class_path, nx_class in class_map.items():
+        if f"/{class_path}/" in path or path.startswith(f"{class_path}/"):
+            path = path.replace(f"{class_path}/", f"{nx_class}[{class_path}]/")
+    return path
+
+
+def _clean_str_attr(attr: Optional[Union[str, bytes]], encoding="utf-8") -> str:
+    if attr is None:
+        return attr
+    if isinstance(attr, bytes):
+        return attr.decode(encoding)
+    if isinstance(attr, str):
+        return attr
+
+    raise TypeError(
+        "Invalid type {type} for attribute. Should be either None, bytes or str."
+    )
+
+
+def _get_def_map(file: str) -> Dict[str, str]:
+    def_map: Dict[str, str] = {}
+    with File(file, "r") as h5file:
+        for entry_name, dataset in h5file.items():
+            if _clean_str_attr(dataset.attrs.get("NX_class")) == "NXentry":
+                def_map = {
+                    entry_name: (
+                        definition := h5file[f"/{entry_name}/definition"][()].decode(
+                            "utf8"
+                        )
+                    )
+                }
+                logger.debug("Reading entry '%s': '%s'", entry_name, definition)
+
+    return def_map
+
+
+@click.command()
+@click.argument("file")
+def verify(file: str):
+    """Verifies a nexus file"""
+    def_map = _get_def_map(file)
+
+    if not def_map:
+        logger.info("Could not find any valid entry in file %s", file)
+    ref_template = Template()
+    data_template = Template()
```

Comment:
Just a side note: Can these lines be ommitted?

Reply:
Yes they can but linting complained that I initialized them in the for-loop. Anyways, I will clean this all up a little before actually merging it. It's was more of a proof-of-concept now.